File: sol/386/386.cpp
struct Solution {
    vector<int> lexicalOrder(int n) {
        vector<int> res(n);
        int val = 1;
        for (int i = 0; i < n; ++i){
            res[i] = val;
            if (val * 10 <= n){
                val *= 10;
            } else {
                while (val % 10 == 9 || val + 1 > n){
                    val /= 10;
                }
                ++val;
            }
        }
        return res;
    }
};

File: sol/386/386.py
class Solution:
    def lexicalOrder(self, n: int) -> List[int]:
        res = [0] * n
        val = 1
        for i in range(n):
            res[i] = val
            if (p := val * 10) <= n:
                val = p
            else:
                while val % 10 == 9 or val + 1 > n:
                    val //= 10
                val += 1
        return res
